src/app/clusters/ota-requestor/GenericOTARequestorDriver.cpp

```diff
@@ -146,14 +147,17 @@ void GenericOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason, Syst
         break;
     case UpdateNotFoundReason::kBusy:
         willTryAnotherQuery = true;
-        break;
+        if (mProviderRetryCount <= kMaxBusyProviderRetryCount)
+        {
+            break;
+        }
+        ChipLogProgress(SoftwareUpdate, "Max Busy Provider retries reached. Attempting to get next Provider.");
+        __attribute__((fallthrough)); // fallthrough
     case UpdateNotFoundReason::kNotAvailable: {
```

Comment:
Is there also going to be a max retry for this case or is this for a later PR?

Reply:
Resolved conversation over Slack huddle.